### PR TITLE
Added additional EventHub trigger samples

### LINKED ISSUE #1354: Enhance EventHub samples to include multiple binding scenarios
There was a recent issue filed (#1256) in which a customer used the EventHub samples as an example to change their parameter from EventData[] to byte[], and was having trouble making it work. 

We want to add samples for both batched and non-batched scenarios, detailed [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/1256#issuecomment-1414693442).

### PR DESCRIPTION
**Pull request by madelinegordon**
### Issue describing the changes in this PR

resolves #1354

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Added EventHub triggered samples to account for scenarios where messages are dequeued as `IEnumerable<byte[]>` and `byte[]`.

### CODE REVIEW

**Review by liliankasem — APPROVED** [review]

**Review by satvu — APPROVED** [review]

File: samples/Extensions/EventHubs/EventHubsTriggerMetadata.cs [hunk 1]
```diff
@@ -50,5 +52,44 @@ public static void UsingContext([EventHubTrigger("src-context", Connection = "Ev
                 Dictionary<string, JsonElement> systemProperties = systemPropertiesArray[i];
             }
         }
+
+        // batched scenarios
+        [Function("EventHubsTriggerMetadata-ByteParameter-Batched")]
+        public static void UsingByteParameterBatched([EventHubTrigger("src-parameters", Connection = "EventHubConnectionAppSetting")] IEnumerable<byte[]> batchedMessages,
```
Comment (kshyju) [inline samples/Extensions/EventHubs/EventHubsTriggerMetadata.cs:58]: We might have a 🐛 in the binding pipeline which causes batchedMessages to be null.

**Comment by liliankasem** [comment]
@madelinegordon any updates on the state of this PR? If it's blocked you can add a blocked tag till its ready

**Comment by fabiocav** [comment]
Investigating issues with the binding type, will reopen the PR when that is resolved.

File: samples/Extensions/EventHubs/EventHubsTriggerMetadata.cs (hunk 1 above)
Comment (ynnova) [reply samples/Extensions/EventHubs/EventHubsTriggerMetadata.cs:58]: Is the issue solved? there's a workaround?

**Comment by andhallberg** [comment]
@fabiocav Is there an issue we can track for the `IEnumerable<byte[]>` being null when triggered? We are facing this issue when having switched from in-process to isolated process.

**Pull request closed without merging**